### DISCUSSION TITLE
Is the tail-call feature really needed?

### DISCUSSION THREAD

**Discussion by ChrML** [sebastienros/jint#1926] [Q&A]
I see that the README doc marks tail-calls red as one of the two missing features to comply with ES5. However it seems almost no JS engines today support it, and it was removed from V8. So is there really a need to add this complexity to Jint?

It seems status quoi is that as a JS developer, even tho it's been since ES5, you can't really rely your JS code on the engine having tail-calls. And there's lots of arguments about it not being so useful .. / narrow use-case / doesn't help performance.

Question is, will it be implemented, or should the readme just be changed to indicate that there is no intension to implement it?

**Comment by lahma** [comment]
The readme tries to be a feature sheet which shows which parts have been implemented and the current status. I don't know if it would make much difference if there would be a not that it won't be implemented. Maybe someone brave wants to implement it someday. On .NET side at least [NiL.JS](https://github.com/nilproject/NiL.JS) does support it.

**Reply by sebastienros** [reply]
Done!
